projects/planner/src/app/diff/diveinfo/results/diff-diveinfo-results.component.ts

```diff
@@ -0,0 +1,164 @@
+        return this.profileComparatorService.profileAResults();
+    }
+
+    public get profileB(): DiveResults {
+        return this.profileComparatorService.profileBResults();
+    }
+
+    public get areResultsCalculated(): boolean {
+        return this.profileComparatorService.areResultsCalculated();
+    }
+
+    public get areDiveInfosCalculated(): boolean {
+        return this.profileComparatorService.areDiveInfosCalculated();
+    }
+
+    public get areProfilesCalculated(): boolean {
+        return this.profileComparatorService.areProfilesCalculated();
+    }
+
+    public get isComplex(): boolean {
+        return this.viewSwitch.isComplex;
+    }
+
+    public get needsReturn(): boolean {
+        // TODO: Add Profile B
+        return this.profileA.needsReturn;
+    }
+
+    public get totalDurationDifference(): number {
+        return this.profileB.totalDuration - this.profileA.totalDuration;
+    }
+
+    public get timeToSurfaceDifference(): number {
+        return this.profileB.timeToSurface - this.profileA.timeToSurface;
+    }
+
+    public get averageDepthDifference(): number {
+        return this.averageDepthOfProfile(this.profileB) - this.averageDepthOfProfile(this.profileA);
+    }
+
+    public get emergencyAscentStartDifference(): number {
+        return this.profileB.emergencyAscentStart - this.profileA.emergencyAscentStart;
+    }
+
+    public get noDecoDifference(): number {
+        return this.noDecoOfProfile(this.profileB) - this.noDecoOfProfile(this.profileA);
+    }
+
+    public get maxTimeDifference(): number {
+        return this.profileB.maxTime - this.profileA.maxTime;
+    }
+
+    public get highestDensityDifference(): number {
+        return this.highestDensityOfProfile(this.profileB) - this.highestDensityOfProfile(this.profileA);
+    }
+
+    public get otuDifference(): number {
+        return this.profileB.otu - this.profileA.otu;
+    }
+
+    public get cnsDifference(): number {
+        return this.profileB.cns - this.profileA.cns;
+    }
+    public get cnsDifferenceText(): string {
+        const diff = this.cnsDifference;
+        if(diff >= 1000) {
+            return '> 1000';
+        }
+
+        if(diff <= -1000) {
+            return '< -1000';
+        }
+
+        return formatNumber(diff, 'en', '1.0-0');
+    }
+
+    public showMaxBottomTimeOfProfile(profile: DiveResults): boolean {
+        return profile.maxTime > 0;
+    }
+
+    public noDecoOfProfile(profile: DiveResults): number {
+        return profile.noDecoTime;
+    }
+
+    public averageDepthOfProfile(profile: DiveResults): number {
+        return this.units.fromMeters(profile.averageDepth);
+    }
+
+    public highestDensityOfProfile(profile: DiveResults): number {
+        const density = profile.highestDensity.density;
+        return this.units.fromGramPerLiter(density);
+    }
+
+    public densityTextOfProfile(profile: DiveResults): string {
+        const gas = profile.highestDensity.gas.name;
```

Comment:
Is the gas always available?

Reply:
Looks like it. DiveResults.highestDensity gets declared & initiated as ` public highestDensity = HighestDensity.createDefault();`, which adds Air at 0 depth with density 0.